content/articles/generic-programming-in-c#/index.md

```diff
@@ -0,0 +1,182 @@
+
+We have three common generic collections incorporated in the C# namespace.
+
+## 1. LIST
+
+A list is the most commonly used generic collection. Its because they are much more convenient when they are used instead of an array.
+A list is a very general data structure with each entry occupying a specific position known as an index. Lists are very flexible as they automatically resize every time data is added.
+
+Lists are very easy to manipulate. This is because many methods have been created and abstracted, meaning the bulky work is already done for the user. The user only has to call the method and manipulate the list.
+ 
+A list stores data in sequence, although the sequence oughtn't to have to be particular. It can also accept duplicate entries.
+Below is a pictorial explanation of a list. A generic list class is declared as shown:
+
+```
+public class genericList<T>
+
+{
+    
+    void add(T input){}
+}
+```
+
+On the same note, a list can be of integer or string data type. They are declared differently but the syntax is the same as shown:
+```
+using system;
+
+{
+private class genericListExample
+static void Main ()
+{
+    genericlist<int> list1 = new genericlist<int>(); //This declares a  list of type int.
+    
+     genericlist<string> list2 = new genericlist<string>(); //This declares a list of type string.
+     
+      genericlist<int> list1 = new genericlist<int>(); //This declares a list of type int.
+}
+}
+```
+
+A simple c# example of generic list implementation might help in understanding.
+
+```
+using system;
+using system.collection.generic; // A system namespace for implementing generic lists.
+
+namespace GenericListExample
+{
+    public class Generic<T>  // Declare Generics.
+    {
+        public void GenFunction(T printvalue)
+        {
+            Console.WriteLine(printvalue)
+        }
+    }
+    public class program
+    {
+        public static void Main(string [] args)
+        {
+            Console.WriteLine("Printing integer value");
+            Generic<int> gen = new Generic<int>();
+            gen.GenFunction(10);
+            
+            Console.WriteLine("Printing string value");
+            Generic<string> genString = new Generic<string>();
+            gen.GenFunction("Hello world");
+            
+        }
+    }
+}
+```
+
+Note that _<T>_ is a placeholder for data types and replaced by datatype at compile time.
+
+### 2. Hashsets
+This is an implementation of a set. A set is like a big bag of entries with no particular order rather than a list. They are based on the discrete mathematic's idea of a 'set', which is the grouping of entries that can contain one, two, or many distinct items.  Sorting in a hash set is impossible because the entries have to be sorted. Use of a hash set is very much applicable where we have to store a large amount of data that doesn't have to be sorted eg storing the identity of cars passing through an entrance gate. A program example:
```

Comment:
> rather than a list

I would probably rephrase it to say *unlike a list*. It just feels like a more appropriate word choice. Although it shouldn’t make too large of a difference.

Reply:
> Sorting in a hash set is impossible because the entries have to be sorted

It’s hard to understand what you are trying to say here. I would revise this. First, you say that sorting a hash set is impossible, but then follow up by saying it must be sorted. That second part sounds like you are saying it’s possible to sort it which is contradictory.

Reply:
> items.  Sorting

There seems to be an extra space here after the period

Reply:
> Use of a hash set is very much applicable where we have to store a large amount of data that doesn't have to be sorted eg storing the identity of cars passing through an entrance gate

I would probably separate the example (the part after the *eg*) into its own sentence. That sentence is rather long. You could at very least put a comma between the first part and the example, but I’d highly recommend making the example its own sentence.

Reply:
> A program example:

I would probably rephrase this to something like:

> Here is an example program to help you understand this concept:

Since *example* is acting as an adjective to *program* which is a noun. Also just saying, *an example program* doesn’t flow well, so I altered it further to connect more with what was being said.